lib/rubocop/cop/lint/nested_method_definition.rb

```diff
@@ -27,7 +27,7 @@ class NestedMethodDefinition < Cop
         PATTERN
 
         def_node_matcher :class_or_module_new_call?, <<-PATTERN
```

Comment:
Maybe `:allowed_def_inside_new_with_block?` ? It's less explicit, but allows further extension without needing another rename.